public/app/features/commandPalette/actions/global.static.actions.ts

```diff
@@ -58,95 +56,50 @@ export default (navBarTree: NavModelItem[]) => {
-          keywords: 'new add',
-          perform: () => locationService.push('/dashboard/new'),
-          section: 'Management',
-        },
-      ],
-    },
-    {
-      url: '/',
-      actions: [
-        {
-          id: 'go/home',
-          name: 'Go to home',
-          keywords: 'navigate',
-          perform: () => locationService.push('/'),
-          section: 'Navigation',
-          shortcut: ['g', 'h'],
-          priority: Priority.HIGH,
-        },
-      ],
-    },
-    {
-      url: '/explore',
-      actions: [
-        {
-          id: 'go/explore',
-          name: 'Go to explore',
-          keywords: 'navigate',
-          perform: () => locationService.push('/explore'),
-          section: 'Navigation',
-          priority: Priority.NORMAL,
-        },
-      ],
-    },
-    ...alertingCommandPaletteStaticActions,
-    {
-      url: '/profile',
-      actions: [
-        {
-          id: 'go/profile',
-          name: 'Go to profile',
-          keywords: 'navigate preferences',
-          perform: () => locationService.push('/profile'),
-          section: 'Navigation',
-          shortcut: ['g', 'p'],
-          priority: Priority.LOW,
-        },
-      ],
-    },
-    {
-      url: '/datasources',
-      actions: [
-        {
-          id: 'go/configuration',
-          name: 'Go to data sources configuration',
-          keywords: 'navigate settings ds',
-          perform: () => locationService.push('/datasources'),
-          section: 'Navigation',
-        },
-      ],
-    },
-  ];
-
-  const navBarActions: Action[] = [];
-  const navBarFlatUrls = flatMapDeep(navBarTree, (nbt) => nbt.children)
-    .map((nbf) => nbf?.url)
-    .filter((url) => url !== undefined);
+  interface NavItemWithParent extends NavModelItem {
+    parent?: NavModelItem;
+  }
 
-  navBarActionMap.forEach((navBarAction) => {
-    const navBarItem = navBarFlatUrls.find((url) => navBarAction.url === url);
-    if (navBarItem) {
-      navBarActions.push(...navBarAction.actions);
+  const flatNavTree = navBarTree.flatMap<NavItemWithParent>((navItem) => {
+    // exclude parent items without URLs because we can't navigate to them
+    if (!navItem.url) {
+      return [];
     }
+
+    const children = (navItem.children ?? []).map((childNavItem) => {
+      const grandchild: NavItemWithParent = {
+        ...childNavItem,
+        parent: navItem,
+      };
+
+      return grandchild;
+    });
+
+    const parent: NavItemWithParent = navItem;
+
+    return [parent, ...children];
   });
 
+  function idFromNavItem(navItem: NavModelItem) {
+    return 'navModelItem.' + (navItem.id ?? navItem.url);
+  }
+
+  const navBarActions = flatNavTree
+    .map<Action | undefined>((navItem) => {
+      const { url, id, text } = navItem ?? {};
+      if (!url || !id || !text) {
+        return;
+      }
+
+      return {
+        id: idFromNavItem(navItem),
+        name: `Go to ${text}`,
+        perform: () => locationService.push(url),
```

Comment:
Putting `perform` on parent nav tree actions (like "Go to Administration") in the screenshot prevent the user from being able to view that item's child actions in the command palette because instead it will navigate to them. I think that's fine?